core/src/main/java/com/klaviyo/coresdk/networking/NetworkBatcher.kt

```diff
@@ -0,0 +1,67 @@
+package com.klaviyo.coresdk.networking
+
+import android.os.Handler
+import android.os.HandlerThread
+import com.klaviyo.coresdk.KlaviyoConfig
+import java.util.concurrent.ConcurrentLinkedQueue
+
+object NetworkBatcher {
+    private val handlerThread = HandlerThread("Klaviyo Network Batcher")
+    private var handler: Handler? = null
+    private var batchQueue = ConcurrentLinkedQueue<NetworkRequest>()
+    private var queueInitTime = 0L
+
+    internal fun initBatcher() {
+        if (!handlerThread.isAlive) {
+            handlerThread.start()
+            handler = Handler(handlerThread.looper)
+        }
+        handler?.post(NetworkRunnable())
+    }
+
+    internal fun forceEmptyQueue() {
+        handler?.removeCallbacksAndMessages(null)
+        handler?.post(NetworkRunnable(true))
+    }
+
+    internal fun getBatchQueueSize(): Int {
+        return batchQueue.size
+    }
+
+    internal fun batchRequests(vararg requests: NetworkRequest) {
+        if (batchQueue.size == 0) {
+            initBatcher()
+            queueInitTime = System.currentTimeMillis()
+
+        }
+
+        for (request in requests) {
+            if (request is TrackRequest) {
+                request.generateUnixTimestamp()
+            }
+            batchQueue.add(request)
```

Comment:
if this is a queue, should be be using the queue interface over this? ie use `.offer` instead of `.add`, i think the source code for `ConcurrentLinkedQueue` also suggests using offer instead of add

Reply:
I'll update sure cause I think its harmless to do so. The difference is super minimal though to my understanding. `.offer` will give us some exception handling for the cases where we can't add an element to the queue, but since our queue is unbounded (we have a threshold for checking if we should empty the queue of requests but not a bounded size constraint cause we still want to accept any requests in the interim) it's not actually something that should realistically happen to us.